pywr/_core.pyx

```diff
@@ -962,18 +962,23 @@ cdef class Storage(AbstractStorage):
             except ZeroDivisionError:
                 self._current_pc[i] = np.nan
 
-    cpdef after(self, Timestep ts):
+    cpdef after(self, Timestep ts, double[:] adjustment = None):
         AbstractStorage.after(self, ts)
         cdef int i
         cdef double mxv, mnv
         cdef ScenarioIndex si
 
         for i, si in enumerate(self.model.scenarios.combinations):
             self._volume[i] += self._flow[i]*ts.days
-            # Ensure variable maximum volume is taken in to account
+            # Apply any storage adjustment if given
+            if adjustment is not None:
+                self._volume[i] += adjustment[i]
 
+            # Ensure variable maximum volume is taken in to account
             mxv = self.get_max_volume(si)
             mnv = self.get_min_volume(si)
+            # Ensure volume stays within bounds.
+            self._volume[i] = min(mxv, max(mnv, self._volume[i]))
```

Comment:
Note to self. This is probably a bad idea from as mass balance point of view. At least make it optional / configurable.

Reply:
@Batch21 did we discuss / resolve my previous concerns about this line?

Reply:
yeah, I think 40aefe00589d2bde48bf3b1e797f67be0e483370 was the result of that discussion